File: backend/questions/entity/question.go
package entity

import "fmt"

// Simple Question struct that just handles the data, never processes.
type Question struct {
	Question      string        `json:"question" bson:"question"`
	Choices       []interface{} `json:"choices,omitempty" bson:"choices,omitempty"`
	CorrectChoice interface{}   `json:"correctChoice" bson:"correctChoice"`
}

func (q Question) String() string {
	return fmt.Sprintf("Question: %v\n\t Answer: %v\n", q.Question, q.CorrectChoice)
}

File: frontend/src/app/question/question.ts
import { QuestionType } from './question-type.enum';

export interface Question {
  question: string;
  choices?: Array<string>;
  correctChoice: string | boolean;
  selectedChoice?: string | boolean;
  // if the correctChoice is false this is the correct answer
  answer?: string;
  type: QuestionType;

  isCorrect?: boolean;
}
